text/trace/0220-messaging-semantic-conventions-span-structure.md

```diff
@@ -0,0 +1,590 @@
+conventions outlined above. Green boxes denote spans that are required to
+exist in order to conform to those conventions. Other boxes denote spans that
+are not required and covered by the conventions, but are hopefully helpful in
+understanding how messaging spans can be integrated into an overall trace flow.
+
+### Single message producer, single message push-based consumer
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  PM1[Publish m1]
+  end
+  subgraph CONSUMER
+  DM1[Deliver m1]
+  end
+  PM1-. link .->DM1;
+
+  classDef normal fill:green
+  class PM1,DM1 normal
+  linkStyle 0 color:green,stroke:green
+```
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  PM1[Publish m1]
+  end
+  subgraph CONSUMER
+  DM1[Deliver m1]
+  end
+  PM1-. link .->DM1;
+  PM1-->DM1;
+
+  classDef normal fill:green
+  class PM1,DM1 normal
+  linkStyle 0,1 color:green,stroke:green
+```
+
+### Single message producer, single message push-based consumer with manual settlement
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+    direction TB
+    PM1[Publish m1]
+  end
+  subgraph CONSUMER
+    direction TB
+    DM1[Deliver m1]-->S1[Settle m1]
+  end
+  PM1-. link .->DM1;
+
+  classDef normal fill:green
+  class PM1,DM1,S1 normal
+  linkStyle 0,1 color:green,stroke:green
+```
+
+### Single message producer, single message push-based consumer with auto-settlement
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction TB
+  PM1[Publish m1]
+  end
+  subgraph CONSUMER
+  direction TB
+  DM1[Deliver m1]-->S1[Settle m1]
+  end
+  PM1-. link .->DM1;
+
+  classDef normal fill:green
+  class PM1,DM1,S1 normal
+  linkStyle 0,1 color:green,stroke:green
+```
+
+### Batch message producer with "Create" spans, single message pull-based consumer
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction TB
+  P[Publish]-->CM1[Create m1]
+  P-->CM2[Create m2]
+  end
+  subgraph CONSUMER
+  direction TB
+  RM1[Receive m1]
+  RM2[Receive m2]
+  end
+  CM1-. link .->RM1;
+  CM2-. link .->RM2;
+
+  classDef normal fill:green
+  class P,CM1,CM2,RM1,RM2 normal
+  linkStyle 0,1,2,3 color:green,stroke:green
+```
+
+### Batch message producer, single message push-based consumer
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction TB
+  P[Publish]
+  end
+  subgraph CONSUMER
+  direction TB
+  DM1[Deliver m1]
+  DM2[Deliver m2]
+  end
+  P-. link .->DM1;
+  P-. link .->DM2;
+
+  classDef normal fill:green
+  class P,DM1,DM2 normal
+  linkStyle 0,1 color:green,stroke:green
+```
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction TB
+  P[Publish]
+  end
+  subgraph CONSUMER
+  direction TB
+  DM1[Deliver m1]
+  DM2[Deliver m2]
+  end
+  P-. link .->DM1;
+  P-. link .->DM2;
+  P-->DM1;
+  P-->DM2;
+
+  classDef normal fill:green
+  class P,DM1,DM2 normal
+  linkStyle 0,1,2,3 color:green,stroke:green
+```
+
+### Batch message producer with "Create" spans populated before publish, single message pull-based consumer
+
+```mermaid
+flowchart LR;
```

Comment:
I guess if we go with @lmolkova 's suggestion, we need a link on publish for all the creates, right?

Reply:
Added.